File: app/src/main/java/com/jpmorgan/typicode/pojos/Album.java
package com.jpmorgan.typicode.pojos;

import com.google.gson.annotations.SerializedName;

public class Album {
    @SerializedName("userId")
    private int userId;

    @SerializedName("id")
    private int id;

    @SerializedName("title")
    private String title;

    public void setUserId(int userId){
        this.userId = userId;
    }
    public int getUserId(){
        return this.userId;
    }
    public void setId(int id){
        this.id = id;
    }
    public int getId(){
        return this.id;
    }
    public void setTitle(String title){
        this.title = title;
    }
    public String getTitle(){
        return this.title;
    }

    @Override
    public String toString() {
        return "Album{" +
                "userId=" + userId +
                ", id=" + id +
                ", title='" + title + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Album album = (Album) o;

        if (userId != album.userId) return false;
        if (id != album.id) return false;
        return title != null ? title.equals(album.title) : album.title == null;
    }

    @Override
    public int hashCode() {
        int result = userId;
        result = 31 * result + id;
        result = 31 * result + (title != null ? title.hashCode() : 0);
        return result;
    }
}

File: app/src/main/java/com/jpmorgan/typicode/pojos/Album.kt
package com.jpmorgan.typicode.pojos

import com.google.gson.annotations.SerializedName

class Album : Comparable<Album> {

    @SerializedName("userId")
    var userId: Int = 0

    @SerializedName("id")
    var id: Int = 0

    @SerializedName("title")
    var title: String? = null

    override fun toString(): String {
        return "Album{" +
                "userId=" + userId +
                ", id=" + id +
                ", title='" + title + '\''.toString() +
                '}'.toString()
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val album = o as Album?

        if (userId != album!!.userId) return false
        if (id != album.id) return false
        return if (title != null) title == album.title else album.title == null
    }

    override fun hashCode(): Int {
        var result = userId
        result = 31 * result + id
        result = 31 * result + if (title != null) title!!.hashCode() else 0
        return result
    }

    override fun compareTo(o: Album): Int {
        return this.title!!.compareTo(o.title!!)
    }
}
